glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -166,19 +260,83 @@ open class GleanInternalAPI internal constructor () {
      */
     fun handleBackgroundEvent() {
-        sendPing("baseline")
-        sendPing("events")
+        sendPings(listOf(Pings.baseline, Pings.events))
+    }
+
+    /**
+     * Send a list of pings.
+     *
+     * The ping content is assembled as soon as possible, but upload is not
+     * guaranteed to happen immediately, as that depends on the upload
+     * policies.
+     *
+     * If the ping currently contains no content, it will not be sent.
+     *
+     * @param pings List of pings to send.
+     * @return The async Job performing the work of assembling the ping
+     */
+    internal fun sendPings(pings: List<PingType>) = Dispatchers.API.launch {
+        // TODO: 1552264 Move this logic to Rust
+        if (!isInitialized()) {
+            Log.e(LOG_TAG, "Glean must be initialized before sending pings.")
+            return@launch
+        }
+
+        if (!getUploadEnabled()) {
+            Log.e(LOG_TAG, "Glean must be enabled before sending pings.")
+            return@launch
+        }
+
+        var sentPing = false
+        for (ping in pings) {
+            if (sendPing(ping.name)) {
+                sentPing = true
+            } else {
+                Log.d(LOG_TAG, "No content for ping '$ping.name', therefore no ping queued.")
+            }
+        }
+
+        if (sentPing) {
+            PingUploadWorker.enqueueWorker()
+        }
+    }
+
+    /**
+     * Send a list of pings by name.
+     *
+     * Each ping will be looked up in the known instances of [PingType]. If the
+     * ping isn't known, an error is logged and the ping isn't queued for uploading.
+     *
+     * The ping content is assembled as soon as possible, but upload is not
+     * guaranteed to happen immediately, as that depends on the upload
+     * policies.
+     *
+     * If the ping currently contains no content, it will not be sent.
+     *
+     * @param pingNames List of ping names to send.
+     * @return The async Job performing the work of assembling the ping
+     */
+    internal fun sendPingsByName(pingNames: List<String>): Job? {
+        // val pings = pingNames.mapNotNull { pingName ->
+        //     PingType.pingRegistry.get(pingName)?.let {
+        //         it
+        //     } ?: run {
+        //         logger.error("Attempted to send unknown ping '$pingName'")
+        //         null
+        //     }
+        // }
+        // return sendPings(pings)
+        // TODO: 1552264
+        Log.e(LOG_TAG, "sendPingsByName is a stub")
+        return null
     }
 
-    internal fun sendPing(pingName: String) {
-        val queued = LibGleanFFI.INSTANCE.glean_send_ping(
+    private fun sendPing(pingName: String): Boolean {
+        return LibGleanFFI.INSTANCE.glean_send_ping(
             handle,
             pingName,
             (configuration.logPings).toByte()
-        )
-        if (queued != 0.toByte()) {
-            PingUploadWorker.enqueueWorker()
-        }
+        ) != 0.toByte()
```

Comment:
I think we should really introduce some extension function for this: let's do something like [this](https://github.com/mozilla/glean/blob/110e44584de68a960e92542903b0c3e093f3eaa4/glean-core/android/src/main/java/mozilla/telemetry/glean/rust/LibGleanFFI.kt#L16)

Reply:
In another PR we also said that to take in a follow-up. I think it would be fine to take that in one swoop

Reply:
Created [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1552474)

Reply:
Whatever way around we land the PRs, this will need to be adopted then.